gcp_variant_transforms/beam_io/vcf_header_io.py

```diff
@@ -526,29 +505,21 @@ def _format_header_key(self, key):
 
   def _format_number(self, number):
     # type: (int) -> Optional[str]
-    """Returns the string representation of field_count from PyVCF.
-
-    PyVCF converts field counts to an integer with some predefined constants
-    as specified in the vcf.parser.field_counts dict (e.g. 'A' is -1). This
-    method converts them back to their string representation to avoid having
-    direct dependency on the arbitrary PyVCF constants.
+    """Returns the string representation of number field.
 
     Args:
-      number: An integer representing the number of fields in INFO as specified
-        by PyVCF.
+      number: An integer representing the number of fields in INFO
 
     Returns:
-      A string representation of field_count (e.g. '-1' becomes 'A').
+      A string representation of field_count.
 
     Raises:
       ValueError: if the number is not valid.
     """
     if number is None:
       return None
-    elif number >= 0:
+    if number >= 0:
       return str(number)
-    if number in VCF_HEADER_INFO_NUM_FIELD_CONVERSION:
-      return VCF_HEADER_INFO_NUM_FIELD_CONVERSION[number]
```

Comment:
Deleting this extra condition changes the logic, ie without this condition 'A', 'G', and 'R' will raise an exception while it shouldn't!
Here is what I propose:
```
VALID_VCF_HEADER_INFO_NUM_FIELD = [
    'A', # value when number of alternate alleles dictates the count
    'G', # value when number of genotypes dictates the count
    'R', # value when number of alleles and ref dictates the count
]
...
if number in VALID_VCF_HEADER_INFO_NUM_FIELD:
  return number
``` 

If I am right, why our tests did not catch this then? Should we add a test to check this? I already added an item to the [list of issues](#534).

Reply:
Umm, so 'A' >= 0 is True in python. PyVCF had these values as negatives (ie 'A' was -1, 'G' was -2 I think...) so this check was needed to convert integers to strings, and -1 ... -3 to A/G/R. So from that point, this code works fine (it turns strings AND ints to string, so no issue there). However, it should raise an error for non-A/G/R strings, which it wouldn't at them moment, as 'any string' >=0 is True.

Used HEADER_NUMBER instead (since number can be '.' as well. Made to check that number is int. Also removed unnecessary None check - that PySAM returns '.' instead of None.

Also, I'm leaving numer >=0 since we have a test for that, although I'm not sure if it's ever possible to have our vcf header with number < 0 - in theory, I don't think that should ever happen, but just in case, left that condition.